File: Python/statistics/data_distribution_and_sampling/distribution_of_sample_statistics.py
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

if __name__ == "__main__":
    file = "../../data/statistics/loans_income.csv"
    income = pd.read_csv(file)
    income = income['x']

    sample_data = pd.DataFrame({
        'income': income.sample(1000),
        'type': 'Data',
    })
    sample_mean_05 = pd.DataFrame({
        'income': [income.sample(5).mean() for _ in range(1000)],
        'type': 'Mean on 5',
    })
    sample_mean_20 = pd.DataFrame({
        'income': [income.sample(20).mean() for _ in range(1000)],
        'type': 'Mean on 20',
    })

    income = pd.concat([sample_data, sample_mean_05, sample_mean_20])

    graph = sns.FacetGrid(income,
                          col='type',
                          col_wrap=1,
                          height=2,
                          aspect=2)
    graph.map(plt.hist, "income",
              range=[0, 200000],
              bins=40)

    plt.show()

File: R/Scripts/statistics/data_distribution_and_sampling/distribution_of_sample_statistics.R
library(ggplot2)
data <- read.csv("Data/loans_income.csv")
data <- data[["x"]]

sample_data <- data.frame(income=sample(data,1000),type='data_dist')

#sample of means 5 values
sample_mean_05 <- data.frame(
  income = tapply(
    sample(data,1000*5),
    rep(1:1000, rep(5,1000)),
    FUN=mean),
  type="mean_of_5"
  )
#sample of means 20 values
sample_mean_20 <- data.frame(
  income = tapply(
    sample(data,1000*20),
    rep(1:1000, rep(20,1000)),
    FUN=mean),
  type="mean_of_20"
)

income <- rbind(sample_data,sample_mean_05,sample_mean_20)
income$type = factor(
  income$type,
  levels = c("data_dist","mean_of_5","mean_of_20"),
  labels = c("Data", "Mean of 5", "Mean of 20")
)
#histogram
ggplot(income, aes(x=income))+
  geom_histogram(bins=40)+
  facet_grid(type ~ .)


#qq-plot -how close sample is to a specific distribution

norm_sample <- rnorm(200)
qqnorm(norm_sample)
abline(a=0,b=1, col='blue')
 
pois_sample <- rpois(1:100, lambda = 1)#?
qqnorm(pois_sample)
abline(a=0,b=1, col='blue')
